addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -206,47 +276,75 @@ function factory(dependencies) {
 
         /**
          * @private
-         * @returns {string|undefined}
+         * @returns {boolean}
          */
-        _computeFileType() {
-            if (this.type === 'url' && !this.url) {
-                return clear();
-            } else if (!this.mimetype) {
-                return clear();
-            }
-            switch (this.mimetype) {
-                case 'application/pdf':
-                    return 'application/pdf';
-                case 'image/bmp':
-                case 'image/gif':
-                case 'image/jpeg':
-                case 'image/png':
-                case 'image/svg+xml':
-                case 'image/tiff':
-                case 'image/x-icon':
-                    return 'image';
-                case 'application/javascript':
-                case 'application/json':
-                case 'text/css':
-                case 'text/html':
-                case 'text/plain':
-                    return 'text';
-                case 'audio/mpeg':
-                case 'video/x-matroska':
-                case 'video/mp4':
-                case 'video/webm':
-                    return 'video';
-            }
-            if (!this.url) {
-                return clear();
-            }
-            if (this.url.match('(.png|.jpg|.gif)')) {
-                return 'image';
-            }
-            if (this.url.includes('youtu')) {
-                return 'youtu';
+        _computeIsPdf() {
+            return this.mimetype === 'application/pdf';
+        }
+
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsImage() {
+            if (this.url && this.url.match('(.png|.jpg|.gif)')) {
```

Comment:
> I don't like that an URL is considered an image but I guess this is to follow existing behavior?
Sadly, yes.

> Ideally should not be needed though. If this is only used because "isViewable" (or something else?) needs it, then it should be explicitly done there instead.

It sounds right, will do that.